File: VeloCityServer/VeloCity.Api/Models/Payment.cs
using VeloCity.Api.Models.Enums;

namespace VeloCity.Api.Models;

public class Payment
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public virtual User User { get; set; } = null!;
    public decimal Amount { get; set; } = 0.00m;
    public Currency Currency { get; set; } = Currency.PLN;
    public decimal ExchangeRate { get; set; }
    public decimal AmountInBaseCurrency { get; set; }
    public PaymentMethod PaymentMethod { get; set; } = PaymentMethod.Card;
    public string TransactionId { get; set; }
    public PaymentStatus Status { get; set; } = PaymentStatus.Completed;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: VeloCityMobile/src/types/payment.ts
export type PaymentMethod = 'Card' | 'PayPal' | 'Giftcard';
export type Currency = 'PLN' | 'EUR' | 'USD';
export type PaymentStatus = 'Completed' | 'Pending' | 'Failed';

export interface TopUpBalanceCommand {
  amount: number;
  paymentMethod: PaymentMethod;
  currency: Currency;
}

export interface BalanceDto {
  balance: number;
}

export interface PaymentDto {
  id: number;
  amount: number;
  exchangeRate: number;
  amountInBaseCurrency: number;
  currency: Currency;
  paymentMethod: PaymentMethod;
  transactionId: string;
  status: PaymentStatus;
  createdAt: string;
}

export interface PaymentDtoPaginatedList {
  items: PaymentDto[];
  pageNumber: number;
  pageSize: number;
  totalCount: number;
  totalPages: number;
  hasPreviousPage: boolean;
  hasNextPage: boolean;
}
